compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposePanel.desktop.kt

```diff
@@ -77,6 +77,45 @@ class ComposePanel @ExperimentalComposeUiApi constructor(
     private val clipMap = mutableMapOf<Component, ClipComponent>()
     private var content: (@Composable () -> Unit)? = null
 
+    /**
+     * Represents whether [ComposePanel] is attached to Swing hierarchy.
+     */
+    private var isAttached = false
+
+    /**
+     * Determines whether the Compose state in [ComposePanel] should be disposed
+     * when panel is detached from Swing hierarchy (when [removeNotify] is called).
+     *
+     * If it is set to false, it is developer's responsibility to call [dispose] function
+     * when Compose state and all related to [ComposePanel] resources are no longer needed.
+     * It can be useful for cases when [ComposePanel] can be attached/detached to Swing hierarchy multiple times,
+     * so with [isDisposeOnRemove] = `false` state will be preserved.
+     *
+     * On the other hand, [isDisposeOnRemove] = `true` can be useful for stateless components,
+     * that can be recreated for each attaching to Swing hierarchy.
+     *
+     * @see dispose
+     */
+    @ExperimentalComposeUiApi
+    var isDisposeOnRemove: Boolean = true
+
+    /**
+     * Disposes Compose state and rendering resources.
+     *
+     * Should be called only when [ComposePanel] is detached from Swing hierarchy.
+     * Otherwise, nothing will happen.
```

Comment:
I am a bit afraid with the race that can happen here, so a client will call `dispose` when component is still in the hierarchy. So it won't be disposed at all.
But if we dispose it when it is in the hierarchy, then errors will happen with Compose/Rendering.
I thought about protection like "disposeOnNextRemove" but it makes logic much more complicated

Reply:
Can we just check `if (bridge != null) {` and always dispose?

Reply:
In this case, the error may appear that swing can call paint but it bridge is disposed. 
We can just write it in doc and dispose anyway

Reply:
> In this case, the error may appear that swing can call paint but bridge is disposed.

if we allow to dispose before remove, calling paint is valid, and we shouldn't throw an exception, and just check for null and draw a transparent panel.

Alternative to all that - add `check(!isAttached)` into `dispose()`

Reply:
Ahh, sorry, my mistake. We remove bridge from JLayeredPane, so it won't be drawn. We can skip this check with isAttached

Reply:
Removed